### PR TITLE
[SYCL] Minor host task corrections

### PR DESCRIPTION
**Pull request by KarachunIvan**
1. Fixed compilation error which appeared in case of usage
get_native_mem method with accessor with host target.
2. Exceptions which raised in host task reported as async exceptions.

Signed-off-by: Ivan Karachun <ivan.karachun@intel.com>

### CODE REVIEW

**Comment by vladimirlaz** [comment]
@KarachunIvan, please fix regression

File: sycl/source/detail/queue_impl.cpp [hunk 1]
```diff
@@ -192,11 +192,18 @@ void queue_impl::wait(const detail::code_location &CodeLoc) {
...
+    std::lock_guard<mutex_class> Guard(MMutex);
+    for (std::weak_ptr<event_impl> &EventImplWeakPtr : MEvents)
+      if (std::shared_ptr<event_impl> EventImplPtr = EventImplWeakPtr.lock())
+        Events.push_back(EventImplPtr);
+  }
+
+  for (std::shared_ptr<event_impl> &Event : Events) {
+    Event->wait(Event);
   }
+
```
Comment (s-kanaev) [inline sycl/source/detail/queue_impl.cpp:206]: What's achieved by this change?
Is it required for the fix of the issue? If not, I suggest moving it to a separate patch.

Comment (KarachunIvan) [reply sycl/source/detail/queue_impl.cpp:206]: The change fixes the situation which leads to deadlock which I encountered:
1) queue::wait call
2) queue's mutex MMutex is locked 
3) host task throws an exception which should be reported as asynchronous.
4) reportAsyncException is called
5) thread encounters MMutex in this function which is locked
6) ???
7) _Deadlock!_

Comment (s-kanaev) [reply sycl/source/detail/queue_impl.cpp:206]: I believe this deadlock should also be represented in description of this patch.

Comment (s-kanaev) [reply sycl/source/detail/queue_impl.cpp:206]: Is it sporadic deadlock or stable one?
Either way, we ought to have a test which fails without your patch and succeeds with your patch included.

Comment (KarachunIvan) [reply sycl/source/detail/queue_impl.cpp:206]: It was a stable deadlock. The patch includes a test which hangs without this change.

**Comment by KarachunIvan** [comment]
Had to force-push a commit because of updated commit message.

**Comment by bader** [comment]
> Had to force-push a commit because of updated commit message.

You can use PR title and description for that.

File: sycl/source/detail/queue_impl.cpp (hunk 1 above)
Comment (s-kanaev) [reply sycl/source/detail/queue_impl.cpp:206]: Is it `test6` in `sycl/test/host-interop-task/interop-task.cpp`? I believe this test is only for host-task's exception reproting. Please add a comment describing the test.

Comment (KarachunIvan) [reply sycl/source/detail/queue_impl.cpp:206]: Comment has been added.
